plugins/inventory/gql_inventory.py

```diff
@@ -249,16 +273,26 @@ def add_variable(self, host: str, var: str, var_type: str):
         """
         self.inventory.set_variable(host, var_type, var)
 
-    def add_ipv4_address(self, device):
-        """Add primary IPv4 address to host."""
-        if device["primary_ip4"]:
-            if not device["primary_ip4"].get("host"):
-                self.display.error("Mapping ansible_host requires primary_ip4.host as part of the query.")
-                self.add_variable(device["name"], device["name"], "ansible_host")
-                return
-            self.add_variable(device["name"], device["primary_ip4"]["host"], "ansible_host")
+    def add_ip_address(self, default_ip_version, device):
+        """Add primary IP address to host."""
+        # Check to see what the primary IP host addition is, first case is IPv6
+        order_of_preference = ["primary_ip6"]
+
+        # if default_ip_version is IPv4, prepend, else postpend
+        if default_ip_version.lower() == "ipv4":
+            order_of_preference.insert(0, "primary_ip4")
         else:
-            self.add_variable(device["name"], device["name"], "ansible_host")
+            order_of_preference.append("primary_ip4")
+
+        # Check of the address types in the order preference and if it find the first one, add that primary IP to the host
+        for address_type in order_of_preference:
+            if device.get(address_type, {}).get("host"):
```

Comment:
I don't want us to fall victim to the same issue as #347, so can we please change the logic to account if `device[address_type] = None` as well?

Reply:
We set `order_of_preference` to a list with at a minimum of `["primary_ip4"]. So it cannot be `None`.  I did swap around the order in the next commit to better reflect what the default is.

Reply:
No, that's not what I meant. Something like this:

```python
device = {"foo": "bar", "primary_ip4": None}
device.get("primary_ip4")  # None
device.get("primary_ip4", {})  # still None since the key exists
device.get("primary_ip4", {}).get("host")  # None.get(), raises error
```

Reply:
```suggestion
            if address_type in device and device[address_type].get("host"):
```